packages/permissions/src/utils/has-permissions.ts

```diff
@@ -313,10 +319,17 @@ export const hasSomeDataFence = (options: TOptionsForAppliedDataFence) => {
       // we try to find if the demanded dataFence is available inside the actual datafences
       const specificActualDataFence = Object.entries(
         actualDataFenceByTypeAndGroup
-      ).find(
-        ([dataFenceName, dataFenceValue]) =>
-          dataFenceValue && dataFenceName === toCanCase(demandedDataFence.name)
-      );
+      ).find(([dataFenceName, dataFenceValue]) => {
+        // Either the manage permission matches or the
+        // a demanded view permission is implied by a manage permission.
+        const doesDataFenceMatchByName =
+          dataFenceName === toCanCase(demandedDataFence.name) ||
+          dataFenceName === toCanCase(toManageCase(demandedDataFence.name));
```

Comment:
The fix. The above is just moving logic around to be reusable.

Reply:
to clarify:

```
demandedDF: VIEW
dataFenceName VIEW
output: true

demandedDF: MANAGE
dataFenceName VIEW
output: false

demandedDF: VIEW
dataFenceName MANAGE
output: true

demandedDF: MANAGE
dataFenceName MANAGE
output: true
```

Reply:
Yes.